### PR TITLE
Earth- Ana C14 

### PR DESCRIPTION
**Pull request by anakp07**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails |  It is our database
Describe in your own words what the Controller is doing in Rails | It is how View and Model communicate with each other. 
Describe in your own words what the View is doing in Rails | It renders HTML from the controller. 
Describe an edge-case controller test you wrote | Expecting the path to redirect if an invalid ID is entered into 'update'. 
What is the purpose of using strong params? (i.e. the params method in the controller) |  To dry up the code. 
How are Rails migrations related to Rails models? | Rail migrations change aspects of our models. 
Describe one area of Rails that are still unclear on | Testing, partial views and strong parameter usage.

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      return
+    elsif
+    @task.update(completed_at: Time.now.to_s)
+      redirect_to root_path
+      return
+    else
+      @task.update(completed_at: nil)
+      redirect_to tasks_path
+      return
+      end
```
Comment (dHelmgren) [inline app/controllers/tasks_controller.rb:96]: oops! keep an eye on your indentation!

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        name: params[:task][:name],
+        description: params[:task][:description],
+        completed_at: params[:task][:completed_at]
+    )
+    redirect_to root_path
+    return
+  else
+    render :edit
+    return
+    end
```
Comment (dHelmgren) [inline app/controllers/tasks_controller.rb:62]: looks like your indentation got messed up here
```suggestion
    elsif @task.update(
        name: params[:task][:name],
        description: params[:task][:description],
        completed_at: params[:task][:completed_at]
        )
        redirect_to root_path
        return
    else
        render :edit
        return
    end
```

**Review by dHelmgren — COMMENTED** [review]
# Task List




## Major Learning Goals/Code Review





| Criteria                                           | yes/no, and optionally any details/lines of code to reference |
| -------------------------------------------------- | ------------------------------------------------------------- |
| At least 6 commits with meaningful commit messages | ✔️                                                            |
| Routes follow RESTful conventions                  | ✔️                                                            |
| Uses named routes (like `_path`)                   | ✔️                                                           |
| Creates Models and migrations                      | ✔️                                                            |
| Creates styled views                               | ✔️                                                            |
| Handles errors like nonexistant tasks              | ✔️                                                            |
| Uses `form_with` to render forms in Rails          | ✔️                                                          |

## Functional Requirements/Manual Testing

| Functional Requirement                                                         | yes/no |
| ------------------------------------------------------------------------------ | ------ |
| Successfully handles index & show                                              | ✔️     |
| index & show tests pass                                                        | ✔️     |
| Successfully handles: New, Create                                              | ✔️     |
| New, Create tests pass                                                         | ✔️     |
| Successfully handles: Edit, Update                                             | ✔️     |
| Edit, Update tests pass with valid & invalid task ids                          | ✔️     |
| Successfully handles: Destroy, Task Complete                                   | ✔️     |
| Tests for Destroy & Task Complete include tests for valid and invalid task ids | ✔️     |

## Overall Feedback

| Overall Feedback                | Criteria                                                                                                                                                                                      | yes/no |
| ------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------ |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 6+ in Functional Requirements                                                                                                                                            |✔️
| Yellow (Approaches Standards)   | 3+ in Code Review && 5+ in Functional Requirements, or the instructor judges that this project needs special attention                                                                        |
| Red (Not at Standard)           | 0-2 in Code Review or 0-4 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |

Something went funky with your submission, so some of my comments were based on a weird old version of the code you had submitted. IDK what exactly was going on, but in the future please QUADRUPLE CHECK that you are cloning and doing the PR agains C14 :D
